docs/state-diff/README.md

```diff
@@ -0,0 +1,185 @@
+
+Run your function containing the initial set up of your proofs (`counterBed` or `counterBedNamed` in our [example](./test/kontrol/state-diff/proof-initialization.sol)) with `forge script state-diff/proof-initialization.sol:CounterBed --sig counterBed --ffi`. Running it with `forge test` will also work, but only if its name starts with `test`. Notice the `--ffi` flag: we use `mkdir` and `touch` to handle the cases where the state diff files don't yet exist.
+
+### 🧼 Clean the State Diff file 🧼
+
+Currently, the produced state diff JSON is escaped when generated. Run the [`clean_json.py`](test/kontrol/scripts/json/clean_json.py) script on the generated state diff file to unescape it. Example: `python3 test/kontrol/scripts/json/clean_json.py state-diff/StateDiff.json`
+
+### 📜 Scripting all of this 📜
+
+Phew! This is nothing short of a mouthful of steps! In case you don't want to generate the necessary scripts by manally repeating all that we have gone through, we provide the [`record-state-diff.sh`](test/kontrol/scripts/record-state-diff.sh) script automatically generate the JSON files.
+Note that all the relevant environment variables are also present in the script so it can be reused with your project.
+
+## 📝 Write Your Proofs 📝
+
+By now we have generated two files, one containing the recorded state updates of choice and another with the saved names of the relevant addresses. Once we have written the proofs, the state diff JSON will be used by Kontrol to load all the recorded state updates as the initial configuration of the proofs. This means that if we have a test such as the following
+```solidity
+Counter[] public counters;
+
+function setUp() public {
+        for (uint256 i; i <= 9; ++i) {
+            counters.push(new Counter());
+            counters[i].setNumber(i);
+        }
+    }
+
+function test_multiple_counters() public {
+        for(uint256 i; i <= 9; ++i){
+            require(counters[i].number() == i, "Bad number");
+        }
+    }
+```
+we can offload the computation of the `setUp` function (of any function, really) to Foundry, and then provide to Kontrol the file containing what state updates occurred during the execution of `setUp`. Let's see how this transformation is done.
+
+First, we need to run this `setUp` function with the `recordStateDiff` modifier. You can find that function [here](./test/kontrol/state-diff/proof-initialization.sol#L18):
+```solidity
+function counterBedNamed() public recordStateDiff {
+    for (uint256 i; i <= 9; ++i) {
+        counter = new Counter();
+        counter.setNumber(i);
+        save_address(address(counter), string.concat("counter", vm.toString(i)));
+    }
+}
+```
+Notice that we're saving the addresses via `save_address`, not via the `Counter[]`. 
+
+The next step is to use Kontrol's `load-state-diff` command. An example on how to use it can be found in [the provided script](test/kontrol/scripts/record-state-diff.sh). For our running example it can be called as follows
```

Comment:
For consistency:
```suggestion
The next step is to use Kontrol's `load-state-diff` command. An example on how to use it can be found in [the provided script](test/kontrol/scripts/record-state-diff.sh). For our running example it can be called as follows:
```